File: app/lib/domain/models/exercise.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'exercise.freezed.dart';
part 'exercise.g.dart';

@freezed
class Exercise with _$Exercise {
  const factory Exercise({
    required String name,
    required int sets,
    required String reps,
    @Default(90) int restSeconds,
    String? notes,
  }) = _Exercise;

  factory Exercise.fromJson(Map<String, dynamic> json) =>
      _$ExerciseFromJson(json);
}

File: app/android/app/src/main/java/com/gymcoach/gym_coach/domain/model/Exercise.kt
package com.gymcoach.gym_coach.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Exercise(
    val name: String,
    val sets: Int,
    val reps: String,
    @SerialName("rest_seconds") val restSeconds: Int = 90,
    val notes: String? = null
)
